src/main/java/io/cdap/http/AbstractHandlerHook.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2014-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.http;
```

Comment:
Instead of "copying" all and every files from one project that you need (https://github.com/cdapio/netty-http), I suppose to correct some incompatibility with Java 6, in general, I prefer either:
- to propose a change on the original ones (submitting a proposal)
- to include the original project through maven dependency, and then only the changed files within my project, your project files replacing therefore the dependency ones
This will allow to benefit from correction on the remote project and to not duplicate all jobs (doing twice the fixes).

Aren't you agree?

Reply:
There are very few changes (I checked). I propose on your name the following pull request: 
https://github.com/cdapio/netty-http/issues/73
It might be easier than copying and being able to follow the evolution.
I added the update of dependencies and fix one missing buffer release in the test.

Reply:
At least in the meantime, we could only kept the changed files (not to many) and only source ones, not the test ones (and upgrading dependencies in our pom).

-  src/main/java/io/cdap/http/AbstractHttpResponder.java
-  src/main/java/io/cdap/http/NettyHttpService.java
-  src/main/java/io/cdap/http/SSLHandlerFactory.java
-  src/main/java/io/cdap/http/internal/BasicHttpResponder.java
-  src/main/java/io/cdap/http/internal/HandlerException.java
-  src/main/java/io/cdap/http/internal/HttpResourceHandler.java
-  src/main/java/io/cdap/http/internal/HttpResourceModel.java
-  src/main/java/io/cdap/http/internal/ImmutablePair.java
-  src/main/java/io/cdap/http/internal/ParamConvertUtils.java
-  src/main/java/io/cdap/http/internal/PatternPathRouterWithGroups.java

so 10 files instead of 36 files plus test files. It should be easier to follow the evolution until the merge is done.

WDYT?